src/models/LocationModel.js

```diff
@@ -43,14 +65,13 @@ class LocationModel {
   }
 
   get location (): ?string {
-    if (!this._town) {
+    const town = this._postcode ? `${this._postcode} ${this._town}` : this._town
+    if (!town && !this._address && !this._name) {
       return null
     }
-    const withoutAddress = this._postcode ? `${this._postcode} ${this._town}` : this._town
-    if (!this._address) {
-      return withoutAddress
-    }
-    return `${this._address}, ${withoutAddress}`
+    return [this._name, this._address, town]
+      .filter(value => !!value)
+      .join(', ')
```

Comment:
The only semantic change here, is that the name is displayed if available.
This should be discussed, but imo it makes a lot of sense to do so.